$OpenBSD$

Resolve embedded web assets relative to the project root during Cabal builds.

--- apps/common/Web/Embedded.hs.orig
+++ apps/common/Web/Embedded.hs
@@ -2,14 +2,14 @@
 
 module Web.Embedded where
 
-import Data.FileEmbed (embedDir, embedFile)
+import Data.FileEmbed (embedDir, embedFile, makeRelativeToProject)
 import Simplex.Messaging.Server.Web (EmbeddedContent (..))
 
 embeddedContent :: EmbeddedContent
 embeddedContent =
   EmbeddedContent
-    { indexHtml = $(embedFile "apps/common/Web/static/index.html"),
-      linkHtml = $(embedFile "apps/common/Web/static/link.html"),
-      mediaContent = $(embedDir "apps/common/Web/static/media/"),
-      wellKnown = $(embedDir "apps/common/Web/static/.well-known/")
+    { indexHtml = $(embedFile =<< makeRelativeToProject "apps/common/Web/static/index.html"),
+      linkHtml = $(embedFile =<< makeRelativeToProject "apps/common/Web/static/link.html"),
+      mediaContent = $(embedDir =<< makeRelativeToProject "apps/common/Web/static/media/"),
+      wellKnown = $(embedDir =<< makeRelativeToProject "apps/common/Web/static/.well-known/")
     }
